dev/tasks/linux-packages/yum/arrow.spec.in

```diff
@@ -95,38 +141,54 @@ cd cpp/build
   -DARROW_WITH_SNAPPY=ON \
   -DARROW_WITH_ZLIB=ON \
   -DARROW_WITH_ZSTD=ON \
-  -DCMAKE_BUILD_TYPE=$build_type \
+  -DCMAKE_BUILD_TYPE=$cpp_build_type \
   -DPythonInterp_FIND_VERSION=ON \
   -DPythonInterp_FIND_VERSION_MAJOR=3
 make %{?_smp_mflags}
 cd -
 
-%if %{use_glib}
+%if %{use_glib} && !%{use_meson}
 cd c_glib
 ./autogen.sh
 %configure \
   --with-arrow-cpp-build-dir=$PWD/../cpp/build \
-  --with-arrow-cpp-build-type=$build_type \
+  --with-arrow-cpp-build-type=$cpp_build_type \
   --enable-gtk-doc
 sed -i 's|^hardcode_libdir_flag_spec=.*|hardcode_libdir_flag_spec=""|g' libtool
 sed -i 's|^runpath_var=LD_RUN_PATH|runpath_var=DIE_RPATH_DIE|g' libtool
 ld_library_path=$PWD/arrow-glib/.libs/
 ld_library_path=$ld_library_path:$PWD/plasma-glib/.libs/
 ld_library_path=$ld_library_path:$PWD/parquet-glib/.libs/
-ld_library_path=$ld_library_path:$PWD/../cpp/build/$build_type
+ld_library_path=$ld_library_path:$PWD/../cpp/build/$cpp_build_type
 LD_LIBRARY_PATH=$ld_library_path make %{?_smp_mflags}
 cd -
 %endif
 
 %install
+cpp_build_type=release
 cd cpp/build
 make install DESTDIR=$RPM_BUILD_ROOT
 cd -
 
 %if %{use_glib}
 cd c_glib
+  %if %{use_meson}
+pip3 install meson
```

Comment:
Pin meson version so we don't get surprised in the future?

Reply:
We should support the latest Meson.
If Apache Arrow GLib doesn't work with the latest Meson, it should be reported as error. So no pinned Meson is expected.